public/app/features/scenes/core/SceneObjectBase.tsx

```diff
@@ -27,6 +27,7 @@ export abstract class SceneObjectBase<TState extends SceneObjectState = {}> impl
   events = new EventBusSrv();
 
   constructor(state: TState) {
+    // TODO: Key should not belong to state, is technically it should be stable for a scene object
```

Comment:
I looked at moving key from state to top level object prop. The problem is then that we don't have a way to set state from the outside, for the ScenePanelRepeater it wants to clone objects and set key to a predictable value that is the same for each repeat cycle (each refresh) so that each data refresh is not causing unmount and remounts. 

And SceneObject constructor only takes state as argument so moving it would be a bit more messy, we would have to either extends constructor and clone functions to also take in key as argument or set key after constructor/clone call.